crates/compiler/builtins/bitcode/src/str.zig

```diff
@@ -57,18 +58,34 @@ pub const RocStr = extern struct {
         return result;
     }
 
-    pub fn fromByteList(list: RocList) RocStr {
-        // TODO: upon adding string seamless slices, I believe this branch can be changed to bit manipulation.
+    // This requires that the list is non-null.
+    pub fn fromSubList(list: RocList, start: usize, count: usize, update_mode: UpdateMode) RocStr {
+        const start_byte = @ptrCast([*]u8, list.bytes) + start;
```

Comment:
we don't seem to be doing bounds checks here. is that OK?

Reply:
added comment with more detail and made name clearer.